boa_engine/src/module/source.rs

```diff
@@ -0,0 +1,1903 @@
+                                }
+                            }
+
+                            // 4. Return unused.
+                        }),
+                        context,
+                    );
+                }
+                //     iii. If state.[[IsLoading]] is false, return unused.
+                if !state.loading.get() {
+                    return;
+                }
+            }
+        }
+    }
+
+    /// Concrete method [`GetExportedNames ( [ exportStarSet ] )`][spec].
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-getexportednames
+    pub(super) fn get_exported_names(
+        &self,
+        export_star_set: &mut Vec<SourceTextModule>,
+    ) -> FxHashSet<Sym> {
+        // 1. Assert: module.[[Status]] is not new.
+        // 2. If exportStarSet is not present, set exportStarSet to a new empty List.
+
+        // 3. If exportStarSet contains module, then
+        if export_star_set.contains(self) {
+            //     a. Assert: We've reached the starting point of an export * circularity.
+            //     b. Return a new empty List.
+            return FxHashSet::default();
+        }
+
+        // 4. Append module to exportStarSet.
+        export_star_set.push(self.clone());
+
+        // 5. Let exportedNames be a new empty List.
+        let mut exported_names = FxHashSet::default();
+
+        // 6. For each ExportEntry Record e of module.[[LocalExportEntries]], do
+        for e in &self.inner.code.local_export_entries {
+            //     a. Assert: module provides the direct binding for this export.
+            //     b. Append e.[[ExportName]] to exportedNames.
+            exported_names.insert(e.export_name());
+        }
+
+        // 7. For each ExportEntry Record e of module.[[IndirectExportEntries]], do
+        for e in &self.inner.code.indirect_export_entries {
+            //     a. Assert: module imports a specific binding for this export.
+            //     b. Append e.[[ExportName]] to exportedNames.
+            exported_names.insert(e.export_name());
+        }
+
+        // 8. For each ExportEntry Record e of module.[[StarExportEntries]], do
+        for e in &self.inner.code.star_export_entries {
+            //     a. Let requestedModule be GetImportedModule(module, e.[[ModuleRequest]]).
+            let requested_module = self.inner.loaded_modules.borrow()[e].clone();
+
+            //     b. Let starNames be requestedModule.GetExportedNames(exportStarSet).
+            //     c. For each element n of starNames, do
+            for n in requested_module.get_exported_names(export_star_set) {
+                //         i. If SameValue(n, "default") is false, then
+                if n != Sym::DEFAULT {
+                    //             1. If exportedNames does not contain n, then
+                    //                 a. Append n to exportedNames.
+                    exported_names.insert(n);
```

Comment:
```suggestion
            // a. Let requestedModule be GetImportedModule(module, e.[[ModuleRequest]]).
            let requested_module = self.inner.loaded_modules.borrow()[e].clone();

            // b. Let starNames be requestedModule.GetExportedNames(exportStarSet).
            // c. For each element n of starNames, do
            for n in requested_module.get_exported_names(export_star_set) {
                // i. If SameValue(n, "default") is false, then
                if n != Sym::DEFAULT {
                    // 1. If exportedNames does not contain n, then
                    //    a. Append n to exportedNames.
                    exported_names.insert(n);
```
It would be nice to remove the unneeded indent.

Reply:
Made a pass on all spec comments on the `module` mod and removed unneeded indent